presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -191,6 +194,44 @@ protected Node visitShowTables(ShowTables showTables, Void context)
         }
 
         @Override
+        protected Node visitShowGrants(ShowGrants showGrant, Void context)
+        {
+            String catalogName = session.getCatalog().orElse(null);
+            Expression predicate = equal(nameReference("grantee"), new StringLiteral(session.getUser()));
+
+            Optional<String> identity = showGrant.getIdentity();
+            if (identity.isPresent() && !identity.get().equals(session.getUser())) {
+                denyShowGrants(identity.get());
+            }
+
+            Optional<QualifiedName> tableName = showGrant.getTableName();
+            if (tableName.isPresent()) {
+                QualifiedObjectName qualifiedTableName = createQualifiedObjectName(session, showGrant, tableName.get());
+
+                catalogName = qualifiedTableName.getCatalogName();
+
+                Expression tablePredicate = equal(nameReference("table_name"), new StringLiteral(qualifiedTableName.getObjectName()));
+                predicate = logicalAnd(predicate, tablePredicate);
+            }
+
+            if (catalogName == null) {
```

Comment:
I know I said this before, but I still think you should iterate through all of the catalogs and not only show grants for one catalog.